fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationSettings.kt

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Divider
+import org.mozilla.fenix.compose.SwitchWithLabel
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.list.TextListItem
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Firefox Translation settings fragment compose view.
+ *
+ * @param translationSwitchList A list with TranslationSwitchItems that will appear on the fragment.
+ * @param onAutomaticTranslationClicked Callback when the user presses Automatic Translation button.
+ * @param onNeverTranslationClicked Callback when the user presses Never Translation button.
+ * @param onDownloadLanguageClicked Callback when the user presses Download Language button.
+ */
+@Composable
+fun TranslationSettings(
+    translationSwitchList: List<TranslationSwitchItem>,
+    onAutomaticTranslationClicked: () -> Unit,
+    onNeverTranslationClicked: () -> Unit,
+    onDownloadLanguageClicked: () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer1,
+            ),
+    ) {
+        LazyColumn {
+            items(translationSwitchList) { item: TranslationSwitchItem ->
+                SwitchWithLabel(
```

Comment:
I think we need a follow up bug to change `SwitchWithLabel`. Right now, if we go to our Settings and look for an existing setting with a switch, clicking on the row will toggle the switch. I think the current implementation behaves differently from our actual desired use case.

Reply:
I fixed it . It works like in Settings now.

Reply:
Please mention the global change to QA, just so they can be aware for testing other toggles. 

I do sense the `SwitchWithLabel` portion maybe should be a separate patch for this reason too. (Either works for me, but I think we should be especially in case we break other toggles.)

Reply:
Let's actually move the changes with `SwitchWithLabel` into a separate PR. I would be more comfortable pinging others about how this switch should work - whether or not we should even expose the ability to move the toggle or if it  should just be a visual toggle more than anything.

Reply:
https://github.com/mozilla-mobile/firefox-android/pull/4181